libraries/botbuilder-dialogs-adaptive/src/input/ask.ts

```diff
@@ -113,4 +117,11 @@ export class Ask extends SendActivity implements AskConfiguration {
         result.status = DialogTurnStatus.completeAndWait;
         return result;
     }
+
+    protected onComputeId(): string {
+        if (this.activity instanceof ActivityTemplate) {
+            return `Ask[${StringUtils.ellipsis(this.activity.template.trim(), 30)}]`;
+        }
+        return `Ask[${StringUtils.ellipsis(this.activity && this.activity.toString().trim(), 30)}]`;
```

Comment:
Note: Parity w/ .NET. This would previously show as `SendActivity[...` instead of `Ask[...` which confused me greatly during debugging.